contracts/minter/src/contract.rs

```diff
@@ -120,12 +121,19 @@ pub fn instantiate(
         start_time: msg.start_time,
     };
     CONFIG.save(deps.storage, &config)?;
-    MINTABLE_NUM_TOKENS.save(deps.storage, &msg.num_tokens)?;
 
     let token_list = random_token_list(&env, msg.num_tokens)?;
     // Save mintable token ids map
     for token_id in token_list {
-        MINTABLE_TOKEN_IDS.save(deps.storage, token_id, &true)?;
+        let count = increment_tokens(deps.storage)?;
```

Comment:
I made the same comment on Shane's PR, I think  can we avoid saving the counter every time and just do it once the loop ends and use the  index (index,token_id) to save it in TokenInfo? for 10k collections write operations will have some gas costs implications